Modules/Core/Mesh/test/itkTriangleMeshCurvatureCalculatorTest.cxx

```diff
@@ -95,7 +95,7 @@ itkTriangleMeshCurvatureCalculatorTest(int argc, char * argv[])
   curvCalculator->SetCurvatureTypeToGaussian();
   curvCalculator->Compute();
 
-  gaussCurvatureData = curvCalculator->GetGaussCurvatureData();
+  gaussCurvatureData = curvCalculator->GetModifiableGaussCurvatureData();
```

Comment:
@PranjalSahu Is it really necessary to get "modifiable access" to the curvature data via this `gaussCurvatureData` pointer?  If read-only access is sufficient, I think it would be preferable to keep calling the old `GetGaussCurvatureData()`. And then, just declare `gaussCurvatureData` as (smart) pointer-to-const.

Reply:
The following declaration of `gaussCurvatureData` compiles for me:

    using DoubleVectorContainer = typename CurvatureCalculatorType::DoubleVectorContainer;
    itk::SmartPointer<const DoubleVectorContainer> gaussCurvatureData = curvCalculator->GetGaussCurvatureData();


Or simply:

    auto gaussCurvatureData = curvCalculator->GetGaussCurvatureData();

When doing so (either way), it seems to me that the GetModifiable calls are no longer necessary!

Reply:
Yes, the intention it to only support Read access. Will do the suggested changes and test it.

Reply:
itkGetConstObjectMacro is used for access to Gauss curvature data and I have successfully compile it with ITK_FUTURE_LEGACY_REMOVE flag.

Reply:
Thanks for addressing my comments so carefully @PranjalSahu